sound/soc/sof/intel/hda-pcm.c

```diff
@@ -111,7 +111,13 @@ int hda_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
 
 	dmab = substream->runtime->dma_buffer_p;
 
-	hstream->format_val = rate | bits | (params_channels(params) - 1);
+	/*
+	 * Use the codec required format val (which is link_bps adjusted) when
+	 * the DSP is not in use
```

Comment:
the comment is misleading or incorrect. If the DSP is not in use, then the format_val is selected without taking the codec link_pbs into account.

Reply:
I think it is the other way around: if the DSP is not in use we take the codec link_bps instead of the bps between the host and the dummy codec.